packages/frontend/src/App.jsx

```diff
@@ -6,23 +6,26 @@ import useGameContext from 'hooks/useGameContext';
 
 const App = () => {
   const navigate = useNavigate();
+
   const {
     state: { stage },
   } = useGameContext();
 
   useEffect(() => {
     const { path } = routes[stage];
+
     navigate(path);
+    //eslint-disable-next-line
```

Comment:
I think we could check the current browser location within this `useEffect` and only call `navigate` if the current path doesn't match. This would avoid the infinite render loop and allow `navigate` to be included in the dependency array.

Reply:
Makes sense! I think I implemented this correctly.